### DISCUSSION TITLE
How to Initialise datasource programatically with micronaut.

### DISCUSSION THREAD

**Discussion by Praveen-7163** [micronaut-projects/micronaut-data#2960] [Q&A]
Hi Team, 

can you please show any documents or sample code snippets about initialising datasource beans programmatically in java rather than defining them in application.properties file.

And also how to use EntityManager in above case like do we need to specify that EntityManager should point to any datasource..etc

**Comment by graemerocher** [comment]
Here is an example [di-example.zip](https://github.com/micronaut-projects/micronaut-data/files/15487613/di-example.zip)

**Comment by Praveen-7163** [comment]
Hi @graemerocher , thanks the example you gave is working fine, can you please document this if not present  that we need to specify dailect property in application.properties explicitly when creating datasource beans programatically.
